inc/dbmysql.class.php

```diff
@@ -956,6 +956,25 @@ function updateOrDie($table, $params, $where, $message = '') {
       return $res;
    }
 
+   /**
+    * Update a row in the database or insert a new one
+    *
+    * @since 9.4
+    *
+    * @param string $table   Table name
+    * @param array  $params  Query parameters ([:field name => field value)
+    * @param array  $where   WHERE clause
+    *
+    * @return mysqli_result|boolean Query result handler
+    */
+   public function updateOrInsert($table, $params, $where) {
+      $req = $this->request($table, $where);
+      if ($req->count() == 0) {
+         return $this->insertOrDie($table, $params, 'Unable to create new element or update existing one');
+      }
+      return $this->updateOrDie($table, $params, $where, 'Unable to create new element or update existing one');
```

Comment:
This is really dangerous. It should not update more thant the conflicting element. If there is multiple elements that matches the `$where` conditions, I am pretty sure that the developer does not want to update all of them.
Maybe we should only update `if ($req->count() == 1)`.